### PR TITLE
Fix Dev chain balance values.

### PR DESCRIPTION
**Pull request by Kianenigma**
all explained in #2301. 

Will probably either get merged as is, as a quick temp fix (after removing prints) or will be changed to _in-progress_ for a while.

### CODE REVIEW

File: core/peerset/src/slots.rs [hunk 1]
```diff
@@ -221,6 +221,7 @@ mod tests {
 	}
 
 	#[test]
+	#[ignore]
```
Comment (bkchr) [inline core/peerset/src/slots.rs]: Please just merge master^^

Comment (Kianenigma) [reply core/peerset/src/slots.rs]: merged.

File: srml/staking/src/lib.rs [hunk 2]
```diff
@@ -503,10 +503,25 @@ decl_storage! {
...
+					assert!(
+						T::Currency::free_balance(&stash) >= balance,
+						"Stash account must have sufficient funds to stake the claimed amount."
+					);
+					// TODO: #2301
+					// The notation of minimum valuable balance in staking is currently not properly demonstrated in any
+					// parameter in the system. It should be declared, the assertion should change accordingly and it should
+					// influence the value of shift in Staking's `CurrencyToVote`.
+					assert!(
+						balance > BalanceOf::<T>::sa(min_staked_balance),
```
Comment (gavofyork) [inline srml/staking/src/lib.rs]: `>=`, no?

**Comment by gavofyork** [comment]
i guess this is ok for now, but i'd really like to implement multiplication/overflow properly rather than relying on mismatched sizes of votes & balances.

**Comment by Kianenigma** [comment]
> i guess this is ok for now, but i'd really like to implement multiplication/overflow properly rather than relying on mismatched sizes of votes & balances.

agreed but even if we had floating point numbers available (= no worrying about arithmetic), this issue would still be an option. They are related but not 100% the same thing. 

The emphasis here is that if balances less than N bit are deemed to be dust, it should be indicated somewhere as a parameter because a module might depend on it. This is exactly the relationship between Staking and how polkadot implements balance in a way that 32 bits is dust. 

I actually don't have a strong inclination to merge this as-is and prefer discussing it more first (was expecting a bit more feedback on the issue honestly). Maybe this new _parameter_ should be stated somewhere else (e.g. in balances)? 

But if we want a cleaner dev chain asap this can be merged and will not have much of an impact except one new parameter to Staking's genesis. The entire feature is also effectively disabled if the value is `0`.

**Comment by gavofyork** [comment]
If we didn't need to worry about precision then we'd not bother shifting at all and that would all become moot.

**Pull request closed without merging**